backend/internal/controllers/admin.go

```diff
@@ -0,0 +1,199 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi"
+
+	"github.com/TritonSE/words-alive/internal/auth"
+	"github.com/TritonSE/words-alive/internal/database"
+	"github.com/TritonSE/words-alive/internal/models"
+)
+
+type AdminController struct {
+	Admins database.AdminDatabase
+	Auth   auth.Authenticator
+}
+
+// Create admin account, if request is from another admin account
+// Will write the admin id as a response
+func (c *AdminController) CreateAdmin(rw http.ResponseWriter, req *http.Request) {
+	var cadmin models.CreateAdmin
+	var admin models.Admin
+
+	// Pull CreateAdmin fields from request body
+	if err := json.NewDecoder(req.Body).Decode(&cadmin); err != nil {
+		writeResponse(rw, http.StatusBadRequest, "bad input!")
+		fmt.Println(err)
+		return
+	}
+
+	// Check for duplicate admin email
+	dupAdmin, err := c.Admins.FetchAdminByEmail(req.Context(), cadmin.Email)
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
```

Comment:
400 please :)

Reply:
Oh wait oops - Aksharan was hitting this 500 and I thought it was because you were returning a 500 when there's a duplicate, misread this totally. Let me look and see if I can figure out why this was actually being hit

Reply:
Ok never mind, this looks good. I wrote a test for duplicate email addresses, can you apply this patch please?

```patch
diff --git a/backend/internal/controllers/admin_test.go b/backend/internal/controllers/admin_test.go
index ff52423..5d5d0c6 100644
--- a/backend/internal/controllers/admin_test.go
+++ b/backend/internal/controllers/admin_test.go
@@ -16,7 +16,7 @@ func TestPostAdminUnauthorized(t *testing.T) {
 	fmt.Print("\n================ START ADMIN TESTS ================\n")
 	fmt.Print("\n---------------- CREATE ADMIN TESTS ----------------\n")

-	body := `{"email": "admin1@test.com", "name": "adam",
+	body := `{"email": "admin1@test.com", "password": "p@ssw0rd!", name": "adam",
     "can_manage_users": false, "can_upload_books": true, "can_delete_books" : true}`
 	// not passing token
 	testutils.MakeAuthenticatedRequest(t, "POST", ts.URL+"/admins", body,
@@ -27,7 +27,7 @@ func TestPostAdminUnauthorized(t *testing.T) {
 func TestPostAdminConflictingPerm(t *testing.T) {
 	var response string

-	body := `{"email": "admin1@test.com", "name": "adam",
+	body := `{"email": "admin1@test.com", "password": "p@ssw0rd!", "name": "adam",
     "can_manage_users": true, "can_upload_books": false,
     "can_edit_books": true, "can_delete_books" : false}`
 	testutils.MakeAuthenticatedRequest(t, "POST", ts.URL+"/admins", body,
@@ -40,7 +40,7 @@ func TestPostAdminConflictingPerm(t *testing.T) {
 func TestPostAdmin(t *testing.T) {
 	var response models.Admin

-	body := `{"email": "admin1@test.com", "name": "adam",
+	body := `{"email": "admin1@test.com", "password": "p@ssw0rd!", "name": "adam",
     "can_manage_users": true, "can_upload_books": true, "can_edit_books": true,
     "can_delete_books" : false}`
 	testutils.MakeAuthenticatedRequest(t, "POST", ts.URL+"/admins", body,
@@ -48,7 +48,7 @@ func TestPostAdmin(t *testing.T) {

 	require.Equal(t, "admin1", response.ID)

-	body = `{"email": "intern1@test.com", "name": "ian",
+	body = `{"email": "intern1@test.com", "password": "p@ssw0rd!", "name": "ian",
     "can_manage_users": false, "can_upload_books": false, "can_delete_books" : false}`

 	testutils.MakeAuthenticatedRequest(t, "POST", ts.URL+"/admins", body,
@@ -57,6 +57,18 @@ func TestPostAdmin(t *testing.T) {
 	require.Equal(t, "intern1", response.ID)
 }

+func TestPostAdminDuplicateEmail(t *testing.T) {
+	body := `{"email": "admin1@test.com", "password": "p@ssw0rd!", "name": "adam",
+    "can_manage_users": true, "can_upload_books": true, "can_edit_books": true,
+    "can_delete_books" : false}`
+
+	var res string
+	testutils.MakeAuthenticatedRequest(t, "POST", ts.URL+"/admins", body,
+		http.StatusBadRequest, &res, "test-token-primary")
+
+	require.Equal(t, "duplicate email", res)
+}
+
 // Test list admins without auth
 func TestListAdminsUnauthorized(t *testing.T) {
 	fmt.Print("\n---------------- GET ADMIN TESTS ----------------\n")

```